synapse/federation/transport/client.py

```diff
@@ -149,21 +160,21 @@ async def send_transaction(self, transaction, json_data_callback=None):
         """
         logger.debug(
             "send_data dest=%s, txid=%s",
-            transaction.destination,
-            transaction.transaction_id,
+            transaction.destination,  # type: ignore
+            transaction.transaction_id,  # type: ignore
```

Comment:
Ugh, can we do something to `Transaction` or `JsonEncodedobject` so we don't have to do this? I'd almost prefer *not* adding type hints to the transaction object...

Reply:
I'll take a look, we do this in a couple of other spots too, but it is far from ideal.

Reply:
I think the solution is to make these real objects. I have a WIP branch that does this, but it needs a bit more and I'd rather not let this bitrot. I think we can:

* Merge this as is and fix-up `Transaction` in the future.
* Remove the type hints for `Transaction` (although this ironically makes it harder to fix this in the future...)
* Leave this PR as is until we can fix `Transaction`.

I'd prefer the first option.

Reply:
See #10542 for the eventual follow-up here.